### PR TITLE
SMSApi not implemented - rest works

### PR DESCRIPTION
**Pull request by Aritzaranburu**
@PatrykTyszka SMSApi not implemented yet, everything else works (users receive default pin to make testing of pin verification possible)

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
```
Comment (PatrykTyszka) [inline app/controllers/sessions_controller.rb]: copy/paste ;) use find_by(email: params[:email).

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      PinGenerator.generate(@user)
```
Comment (PatrykTyszka) [inline app/controllers/users_controller.rb]: Why you do not use callback `after save`?

File: app/controllers/verify_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class VerifyController < ApplicationController
+  skip_before_action :redirect_if_unverified
+
+  def new
+    @user = User.find(params[:user_id])
+  end
+
+  def update
```
Comment (PatrykTyszka) [inline app/controllers/verify_controller.rb]: Please refactor this action.
1. Move everything ( without session ) to the form object.
2. Use errors instead flash.now[:alert]

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  has_secure_password
+
```
Comment (PatrykTyszka) [inline app/models/user.rb]: validate format for both (email and phone number)

File: app/services/pin_generator.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class PinGenerator
```
Comment (PatrykTyszka) [inline app/services/pin_generator.rb]: This class doesn't make sense.

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (Aritzaranburu) [reply app/controllers/sessions_controller.rb]: yeah but it works ;) `find_by` is better?

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (Aritzaranburu) [reply app/controllers/users_controller.rb]: I assumed we don't like callbacks. If you want `after_save` it works for me.

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -1,3 +1,20 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
-end
+
+  before_action :redirect_if_unverified
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
```
Comment (PatrykTyszka) [inline app/controllers/application_controller.rb]: use find_by instead find... this piece of code will raise an error when a user does not exist.

File: app/controllers/verify_controller.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
+class ConfirmationsController < ApplicationController
+  skip_before_action :redirect_if_unverified
+
+  def new
+    @user = User.find(params[:user_id])
```
Comment (PatrykTyszka) [inline app/controllers/verify_controller.rb]: Why did you add this? it's useless.

Comment (Aritzaranburu) [reply app/controllers/verify_controller.rb]: After our last planning, I m closing this PR and starting from scratch. I have new signup setup, I ll push it as soon as I make it work. I spent whole day trying to fix some psql connection issue and wanted to throw my laptop out of the window..
